src/Netflix/Netflix.ts

```diff
@@ -0,0 +1,835 @@
+
+/**
+ * Limitations and gotchas:
+ * - This plugin relies on Netflix's `window.netflix` publicized object.
+ *   This plugin MAY fails if netflix introduce a change to that `window.netflix` object
+ * - Watch command relies on videos on cache. It only queries the show netflix has loaded from
+ *   their service and does not query it directly from the service.
+ *
+ * TODOs:
+ * - Remove ContextPatcher stuff when context duplication caused by PluginBase.util.addContext is fixed
+ * - Remove homophone `"search": "search"` after homophone priority issue is fixed on the Platform-land
+ *   https://discuss.lipsurf.com/t/homophones-must-be-a-lower-priority-than-other-plugins-command/297/3
+ */
+
+/// <reference types="lipsurf-types/extension"/>
+declare const PluginBase: IPluginBase;
+
+// Messaging
+
+const LIPSURF_BOOT_SCRIPT_ID = "lipsurf-netflix-script";
+const TO_PAGE_PROOF_KEY = PluginBase.util.getNoCollisionUniqueAttr();
+const FROM_PAGE_PROOF_KEY = PluginBase.util.getNoCollisionUniqueAttr();
+
+type Message<T> = { proofKey: string; payload: T };
+
+const sendMessage = (payload: Command) =>
+  (window as any).postMessage(
+    JSON.stringify({ proofKey: TO_PAGE_PROOF_KEY, payload } as Message<Command>)
+  );
+
+// Utilities
+
+const parseJsonOrNull = <T>(maybeJSONString: string): Message<T> | null => {
+  try {
+    return JSON.parse(maybeJSONString);
+  } catch {
+    return null;
+  }
+};
+
+const consumeMessageStringAsCommand = (
+  messageStr: string,
+  proofKey: string,
+  callback: (command: Command) => unknown
+) => {
+  const message = parseJsonOrNull<Command>(messageStr);
+  if (message && message.proofKey === proofKey) {
+    callback(message.payload);
+  }
+};
+
+const navigateToSearch = (title: string) => {
+  window.location.href = `https://www.netflix.com/search?q=${encodeURIComponent(
+    title
+  )}`;
+};
+
+const navigateToWatch = (videoId: string) => {
+  window.location.href = `https://www.netflix.com/watch/${videoId}`;
+};
+
+// Context
+
+enum NetflixPluginContextEnum {
+  watch = "Netflix Video Player Controls",
+  browse = "Browse Netflix"
+}
+type NetflixPluginContext = NetflixPluginContextEnum | null;
+
+const contextManager = (() => {
+  let enabled = true;
+
+  const createContextFromUrl = (url: URL): NetflixPluginContext => {
+    const { pathname } = url;
+    switch (true) {
+      case pathname.startsWith("/watch"):
+        return NetflixPluginContextEnum.watch;
+      case pathname.startsWith("/latest"):
+      case pathname.startsWith("/browse"):
+      case pathname.startsWith("/title"):
+      case pathname.startsWith("/search"):
+        return NetflixPluginContextEnum.browse;
+      default:
+        return null;
+    }
+  };
+
+  const setContext = (context: NetflixPluginContext) => {
+    switch (true) {
+      case context === NetflixPluginContextEnum.browse: {
+        PluginBase.util.removeContext(NetflixPluginContextEnum.watch);
+        PluginBase.util.addContext(NetflixPluginContextEnum.browse);
```

Comment:
Can you try using enterContext to be more explicit about the context ordering? Also I think it will fix the "search" homophone issue.

Reply:
It will need a quite complex sequence to ensure `tag` works, though. I'll try using enterContext

Reply:
It works. I'm not sure it is ideal though.